retrofit2/src/main/java/com/linecorp/armeria/client/retrofit2/RetrofitMeterIdPrefixFunction.java

```diff
@@ -0,0 +1,137 @@
+ * <ul>
+ *     <li>{@code service} - Retrofit service interface name or defaultServiceName if Retrofit service interface
+ *                           name is not available</li>　
+ *     <li>{@code method} - Retrofit service interface method name or {@link HttpMethod#name()} if Retrofit
+ *                          service interface name is not available</li>
+ *     <li>{@code httpStatus} - {@link HttpStatus#code()}</li>
+ * </ul>
+ */
+public final class RetrofitMeterIdPrefixFunction implements MeterIdPrefixFunction {
+
+    public static final class RetrofitMeterIdPrefixFunctionBuilder {
+
+        private final String name;
+        @Nullable
+        private String tagName;
+        @Nullable
+        private String defaultServiceName;
+
+        private RetrofitMeterIdPrefixFunctionBuilder(String name) {
+            this.name = name;
+        }
+
+        /**
+         * Make tag of {@link RetrofitMeterIdPrefixFunction} contains Retrofit service interface name.
+         */
+        public RetrofitMeterIdPrefixFunctionBuilder withServiceTag(String tagName, String defaultServiceName) {
+            this.tagName = tagName;
+            this.defaultServiceName = defaultServiceName;
+            return this;
+        }
+
+        public RetrofitMeterIdPrefixFunction build() {
+            return new RetrofitMeterIdPrefixFunction(name, tagName, defaultServiceName);
+        }
+    }
+
+    /**
+     * Creates a {@link RetrofitMeterIdPrefixFunctionBuilder} with {@code name}.
+     */
+    public static RetrofitMeterIdPrefixFunctionBuilder builder(String name) {
+        return new RetrofitMeterIdPrefixFunctionBuilder(requireNonNull(name, "name"));
+    }
+
+    private final String name;
+    @Nullable
+    private final String serviceTagName;
+    @Nullable
+    private final String defaultServiceName;
+
+    RetrofitMeterIdPrefixFunction(String name, @Nullable String serviceTagName,
+                                  @Nullable String defaultServiceName) {
+        this.name = name;
+        if (serviceTagName != null && defaultServiceName != null) {
+            this.serviceTagName = serviceTagName;
+            this.defaultServiceName = defaultServiceName;
+        } else {
+            this.serviceTagName = null;
+            this.defaultServiceName = null;
+        }
+    }
+
+    @Override
+    public MeterIdPrefix activeRequestPrefix(MeterRegistry registry, RequestLog log) {
+        return new MeterIdPrefix(name, buildTags(log));
+    }
+
+    @Override
+    public MeterIdPrefix apply(MeterRegistry registry, RequestLog log) {
+        final List<Tag> tags = buildTags(log);
+        // Add the 'httpStatus' tag.
+        final HttpStatus status;
+        if (log.isAvailable(RequestLogAvailability.RESPONSE_HEADERS)) {
+            status = log.status();
+        } else {
+            status = HttpStatus.UNKNOWN;
+        }
+        tags.add(Tag.of("httpStatus", status.codeAsText()));
+
+        return new MeterIdPrefix(name, tags);
```

Comment:
Not a strong opinion, but how about extracting this method to other class like `MeterIdPrefixUtil` so we can remove the duplicate code in `MeterIdPrefixFunction`?

Reply:
I try to extract as static method, curious do we prefer it as default method?

Reply:
Thanks! I think the static method is fine. 😄